Enable input pointer caching in XNNPACK integration.

kimishpatel opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42840 Enable input pointer caching in XNNPACK integration.**
* #42178 Change quantizer to account for input tensor's memory format.
* #42008 Call qnnpack's conv setup only if input pointer has changed.
* #42007 Refactor qconv to reduce allocations.
* #42006 Simple caching allocator for CPU.

Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

kimishpatel committed:
Enable input pointer caching in XNNPACK integration.

Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

AshkanAliabadi commented on aten/src/ATen/native/xnnpack/Convolution.cpp:
```diff
@@ -201,18 +200,38 @@ Tensor run(
+        padded_input_nhwc.size(Layout::Activation4D::height) ||
+      context.input_width !=
+        padded_input_nhwc.size(Layout::Activation4D::width)
+      ) {
```
_Personal_ opinion:

1) Consider using parentheses liberally.
2) I try to put rvalues (or more pedantically prvalues) on the left side of comparisons.  That way `==` or `!=` misspelled as `=` triggers a compiler error.  Otherwise it is a warning, but it is typical in codebases where a 'warning as error' policy is not enforced to fall into the habit of ignoring warnings.

Please feel free to ignore.  Both of these the pedantic side.

kimishpatel replied:
I strongly agree withe second one. I will make those changes. I weakly agree with first one :).

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]

kimishpatel committed:
Update on "Enable input pointer caching in XNNPACK integration."


Summary:
By caching input/output pointers and input parameters we enable the use
of caching allocator and check if we get the same input/output pointers.
If so we skip setup steps.

Test Plan:
python test/test_xnnpack_integration.py

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D23044585](https://our.internmc.facebook.com/intern/diff/D23044585)

[ghstack-poisoned]
